umbral/pre.py

```diff
@@ -472,26 +454,21 @@ def _open_capsule(capsule: Capsule, receiving_privkey: UmbralPrivateKey,
 
     if check_proof:
         offending_cfrags = []
-        for cfrag in capsule._attached_cfrags:
-            if not cfrag.verify_correctness(capsule):
+        for cfrag in cfrags:
+            if not prepared_capsule.verify_cfrag(cfrag):
                 offending_cfrags.append(cfrag)
 
         if offending_cfrags:
             error_msg = "Decryption error: Some CFrags are not correct"
             raise UmbralCorrectnessError(error_msg, offending_cfrags)
 
-    key = _decapsulate_reencrypted(receiving_privkey, capsule)
+    key = _decapsulate_reencrypted(receiving_privkey, prepared_capsule, cfrags)
     return key
 
 
-def decrypt(ciphertext: bytes, capsule: Capsule, decrypting_key: UmbralPrivateKey,
-            check_proof: bool = True) -> bytes:
-    """
-    Opens the capsule and gets what's inside.
-
-    We hope that's a symmetric key, which we use to decrypt the ciphertext
-    and return the resulting cleartext.
-    """
+def decrypt_original(ciphertext: bytes,
+                     capsule: PreparedCapsule,
+                     decrypting_key: UmbralPrivateKey) -> bytes:
```

Comment:
Can you provide docstrings to both decrypt methods?

Reply:
Done